gpdb/gpinitsystem.go

```diff
@@ -113,9 +133,27 @@ func generateSegmentDirectoryList(whichDir string) string {
 
 func (i *Installation) executeGpInitSystem() {
 	Infof("Executing the gpinitsystem to initialize the database")
+
+	// The defaults to execute the command
+	var args = []string{
+		"-a",
+		"-c",
+		i.GpInitSystemConfigLocation,
+		"-h",
+	}
+
+	//Setup args for single vs. multi segment install
 	if i.SingleORMulti == "multi" {
-		executeOsCommand(fmt.Sprintf("%s/bin/gpinitsystem", os.Getenv("GPHOME")), "-c", i.GpInitSystemConfigLocation, "-h", i.SegmentHostLocation , "-a")
+		args = append(args, i.SegmentHostLocation)
 	} else {
-		executeOsCommand(fmt.Sprintf("%s/bin/gpinitsystem", os.Getenv("GPHOME")), "-c", i.GpInitSystemConfigLocation, "-h", i.HostFileLocation , "-a")
+		args = append(args, i.HostFileLocation)
+	}
+
+	// If postgres.conf file available, add this arg as well
+	if i.checkPGConfigFile() {
```

Comment:
It should go in here the check for DataLabs.

```suggestion
	if Config.CORE.DATALABS && i.checkPGConfigFile() {
```
```